pkg/rancher-desktop/backend/k3sHelper.ts

```diff
@@ -1049,26 +1051,41 @@ export default class K3sHelper extends events.EventEmitter {
    * This exists to work around https://github.com/k3s-io/k3s/issues/5103
    */
   async uninstallTraefik(client: KubeClient) {
-    try {
-      const customApi = client.k8sClient.makeApiClient(CustomObjectsApi);
-      const { body: response } = await customApi.listNamespacedCustomObject('helm.cattle.io', 'v1', 'kube-system', 'helmcharts');
-      const charts: V1HelmChart[] = (response as any)?.items ?? [];
+    const deadline = Date.now() + 10 * 60 * 1_000;
+
+    // If the Kubernetes server is not ready yet, we need to retry until it is.
+    // However, don't attempt that forever; only loop until we hit a deadline.
+    while (Date.now() < deadline) {
+      try {
+        const customApi = client.k8sClient.makeApiClient(CustomObjectsApi);
+        const { body: response } = await customApi.listNamespacedCustomObject('helm.cattle.io', 'v1', 'kube-system', 'helmcharts');
+        const charts: V1HelmChart[] = (response as any)?.items ?? [];
 
-      await Promise.all(charts.filter((chart) => {
-        const annotations = chart.metadata?.annotations ?? {};
+        await Promise.all(charts.filter((chart) => {
+          const annotations = chart.metadata?.annotations ?? {};
 
-        return chart.metadata?.name && (annotations['objectset.rio.cattle.io/owner-name'] === 'traefik');
-      }).map((chart) => {
-        const name = chart.metadata?.name;
+          return chart.metadata?.name && (annotations['objectset.rio.cattle.io/owner-name'] === 'traefik');
+        }).map((chart) => {
+          const name = chart.metadata?.name;
 
-        if (name) {
-          console.debug(`Will delete helm chart ${ name }`);
+          if (name) {
+            console.debug(`Will delete helm chart ${ name }`);
 
-          return customApi.deleteNamespacedCustomObject('helm.cattle.io', 'v1', 'kube-system', 'helmcharts', name);
+            return customApi.deleteNamespacedCustomObject('helm.cattle.io', 'v1', 'kube-system', 'helmcharts', name);
+          }
+        }));
+
+        return;
+      } catch (ex) {
+        if (ex instanceof HttpError && ex.statusCode === 503) {
+          console.debug(`Got Service Unavailable (${ ex.body.message }), retrying...`);
+          await util.promisify(setTimeout)(1_000);
+          continue;
         }
-      }));
-    } catch (ex) {
-      console.error('Error uninstalling Traefik', ex);
+        console.error('Error uninstalling Traefik', ex);
+
+        return;
+      }
```

Comment:
Shouldn't we have another log message that says we have reached the timeout and are giving up? Otherwise you see a couple of "retrying" messages, and then nothing, which looks like it eventually worked.